traitlets/traitlets.py

```diff
@@ -995,6 +1066,7 @@ def setup_instance(*args, **kwargs):
         self._trait_values = {}
         self._trait_notifiers = {}
         self._trait_validators = {}
+        self._static_trait_notifiers = []
```

Comment:
I'm working on a rewrite to remove this. It has to do with complications associated with re-registering tagged notifiers to new traits created via `add_traits`.